### PR TITLE
Upgrade to newer versions of TypeScript and TSLint, add --fix option

### PR DESCRIPTION
**Pull request by emilniklas**
Title speaks for itself. Also, I added a yarn.lock file simply because many people use it, like myself, and it's non destructive for NPM users.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -21,10 +21,6 @@
       "avoid-escape",
       "jsx-double"
     ],
-    "no-unused-variable": [
```
Comment (krzkaczor) [inline tslint.json:24]: why do you remove this rule?

**Review by krzkaczor — COMMENTED** [review]
Looks good. 1 comm

**Comment by krzkaczor** [comment]
Thanks for your contribution :) There is 1 comment and I think we can get this merged.

File: tslint.json (hunk 1 above)
Comment (emilniklas) [reply tslint.json:24]: It is deprecated in TSLint. You're prompted to start using the compilerOption --noUnusedParameters and the like.

Comment (krzkaczor) [reply tslint.json:24]: ahh I see. Okay then, it would be nice to find a substitute for this rule in future...

**Comment by krzkaczor** [comment]
@emilniklas Thanks, awesome work! I will release a new package in few minutes.

**Comment by krzkaczor** [comment]
Hmm, I hope that putting `yarn.lock` in this project won't produce any problems. It should be okay since users shouldn't have another tslint in theirs projects dependencies.

**Comment by emilniklas** [comment]
@krzkaczor Yarn's lockfile should always be included regardless. Yarn doesn't use it for transient dependencies anyway. Check [this](https://yarnpkg.com/en/docs/yarn-lock) out for details :)

**Comment by krzkaczor** [comment]
@emilniklas That's interesting. I was pretty sure that it does https://github.com/yarnpkg/yarn/issues/1067 I will just `npmignore` it then.

**Pull request merged**